packages/synchro-charts/src/components/sc-dial/util.ts

```diff
@@ -0,0 +1,56 @@
+import { DialErrorMessages, DialMessages } from './type';
+
+const DEFAULT_FONT_SIZE = 12;
+const DEFAULT_FONT_FAMILY = 'sans-serif';
+
+export enum ColorConfigurations {
+  BLUE = '#2E72B5',
+  NORMAL = '#3F7E23',
+  WARNING = '#F29D38',
+  CRITICAL = '#C03F25',
+  GRAY = '#D9D9D9',
+  PRIMARYTEXT = '#16191f',
+  SECONDARYTEXT = '#687078',
+  WHITE = '#fff',
+}
+
+export const DefaultDialErrorMessages: DialErrorMessages = {
+  errorTimeLabel: 'Last value at',
+  dataNotNumberError: 'Invalid value',
+};
+
+export const DefaultDialMessages: DialMessages = {
+  error: DefaultDialErrorMessages,
+};
+
+const getTextWidthMap = (mapStr: string): Record<string, number> => {
+  const map: Record<string, number> = {};
+  if (typeof JSON === 'undefined') {
+    return map;
+  }
+  for (let i = 0; i < mapStr.length; i += 1) {
+    const char = String.fromCharCode(i + 32);
+    const size = (mapStr.charCodeAt(i) - 20) / 100;
+    map[char] = size;
+  }
+  return map;
+};
+
+export const getFontWidth = (text: string, size?: number, fontFamily?: string): number => {
+  const defaultWidthMapStr =
+    "007LLmW'55;N0500LLLLLLLLLL00NNNLzWW\\\\WQb\\0FWLg\\bWb\\WQ\\WrWWQ000CL5LLFLL0LL**F*gLLLL5F0LF\\FFF5.5N";
```

Comment:
what is this? I don't understand what is going on with this. wheres this come from? how come when i google search this string it comes up with echarts? Why does this magic text width map get recomputed on every function call? and theres no tests?

Reply:
It's to get the width coefficient for about 95 character.